packages/apps/explorer/src/utils/processChainAccounts.ts

```diff
@@ -0,0 +1,78 @@
+import type { IChainAccounts } from '@/components/AccountBalanceDistribution/components/ChainList';
+
+export interface IViewChain {
+  chainId: string;
+  percentage?: number;
+  balance?: number;
+}
+
+export const calculateMaxChainBalance = (chains: IChainAccounts): number => {
+  return chains.reduce((acc: number, val) => {
+    if (typeof val === 'string') return acc;
+    return val.balance > acc ? val.balance : acc;
+  }, 0);
+};
+
+/**
+ * Checks the percentage of the chainbalance to the maxValue.
+ * maxValue is always 100%
+ * the minvalue to show (when there is a balance) is the THRESHOLD
+ */
+
+const THRESHOLD = 2;
+export const chainBalancePercentage = (
+  balance: number,
+  maxValue: number,
+): number => {
+  if (!balance) return 0;
+  const percentage = (balance / maxValue) * 100;
+  return percentage < THRESHOLD ? THRESHOLD : percentage;
+};
+
+export const processChainAccounts = (
+  chains: IChainAccounts,
+  chainCount: number,
+): IViewChain[] => {
+  const maxChainBalance = calculateMaxChainBalance(chains);
+
+  return Array.from(Array(chainCount).keys()).map((idx) => {
+    const chain = chains.find((c) => {
+      return typeof c !== 'string' && c?.chainId === `${idx}`;
+    });
+
+    if (!chain || typeof chain === 'string') {
+      return {
+        chainId: `${idx}`,
```

Comment:
Would you want to set balance to - and percentage to 0 if there's no account?
Now this has to be determined downstream when you can apply the logic here in one place

Reply:
you want to have the difference between a chain where the user is active and has a balance of 0. and a chain where he is not active: balance: undefined